pyop2/codegen/c/solve.c

```diff
@@ -8,19 +8,53 @@ static PetscBLASInt ipiv_buffer[BUF_SIZE];
 static PetscScalar work_buffer[BUF_SIZE*BUF_SIZE];
 #endif
 
-static void solve(PetscScalar* __restrict__ out, const PetscScalar* __restrict__ A, const PetscScalar* __restrict__ B, PetscBLASInt N)
+#ifndef PYOP2_SOLVE_LOG_EVENTS
+#define PYOP2_SOLVE_LOG_EVENTS
+PetscLogEvent USER_EVENT_solve_memcpy = -1;
+PetscLogEvent USER_EVENT_solve_getrf = -1;
+PetscLogEvent USER_EVENT_solve_getrs = -1;
```

Comment:
I think these want to be static? Or does that not work?

Reply:
Ahh I don't quite remember... I remember that I did have this static and think it was not working, but I wrestled with this stuff for a bit so maybe something else was the problem (rather than the static keyword)